### PR TITLE
CDSID- 643- add packagings

### PR DESCRIPTION
**Pull request by raghu1978**

### CODE REVIEW

File: test/controllers/goodsitems/PackagingControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          status(result) mustBe UNAUTHORIZED
+        }
+      }
+    }
+
+    "return BAD_REQUEST" when {
+
+      "user submits invalid data" in {
+
+        val badData =
```
Comment (kurtlogan) [inline test/controllers/goodsitems/PackagingControllerSpec.scala]: In 619 I created a generator specific lens data type which should help improve how we modify generators on the fly. Have a look when it is in master

File: app/views/goods_items_packagings.scala.html [hunk 2]
```diff
@@ -18,66 +18,44 @@
...
-        <td scope="row">@item.heightMeasure</td>
-        <td scope="row">@item.volumeMeasure.map(_.unitCode)</td>
-        <td scope="row">@item.volumeMeasure.map(_.value)</td>
-   </tr>
+    @components.error_summary(form.errors)
+
+    <h2 class="heading-medium">@messages("packaging.heading")</h2>
+    @if(packagings.nonEmpty) {
+    @if(packagings.size == 1) {
+    <p>@packagings.size @messages("packaging.table.heading")</p>
```
[the remark this answers is not in the document]
Comment (raghu1978) [reply app/views/goods_items_packagings.scala.html]: updated

File: app/views/goods_items_packagings.scala.html [hunk 3]
```diff
@@ -18,66 +18,44 @@
...
+    <p>@packagings.size @messages("packaging.table.multiple.heading")</p>
     }
-    </tbody>
-</table>
-} else {
-<p>No Packagings available</p>
-}
+    }
+
+    @components.table.table(
```
[the remark this answers is not in the document]
Comment (raghu1978) [reply app/views/goods_items_packagings.scala.html]: updated

File: app/views/goods_items_origins.scala.html [hunk 4]
```diff
@@ -18,48 +18,48 @@
...
 @(form:Form[_], origins: Seq[Origin])(implicit request: Request[_], messages: Messages,appConfig:AppConfig)
-
+ @tableHeader = @{
+                if(origins.nonEmpty) {
+                    if(origins.size == 1) {
+                    messages("origin.table.heading")
+                    } else {
+                    messages("origin.table.multiple.heading")
+                    }
+                } else ""
```
Comment (kurtlogan) [inline app/views/goods_items_origins.scala.html]: Don't need else

File: app/views/goods_items_origins.scala.html [hunk 5]
```diff
@@ -18,48 +18,48 @@
 @import uk.gov.hmrc.play.views.html._
 @import controllers.routes._
 @import uk.gov.hmrc.wco.dec.Origin
+@import viewmodels.HtmlTable
 
 @(form:Form[_], origins: Seq[Origin])(implicit request: Request[_], messages: Messages,appConfig:AppConfig)
-
+ @tableHeader = @{
+                if(origins.nonEmpty) {
```
Comment (kurtlogan) [inline app/views/goods_items_origins.scala.html]: Styling is messy here
Don't need to check`origin.nonEmpty`

File: app/views/goods_items_origins.scala.html [hunk 6]
```diff
@@ -18,48 +18,48 @@
...
-        <td scope="row">@item.typeCode</td>
-   </tr>
-    }
-    </tbody>
-</table>
-} else {
-<p>No Origins available</p>
-}
+        @if(origins.isEmpty) {
+            @components.input_text(field = form("countryCode"),label = messages("origin.countryCode"))
```
Comment (kurtlogan) [inline app/views/goods_items_origins.scala.html]: Inputs are the same in both in both branches of the if

File: app/views/goods_items_packagings.scala.html [hunk 7]
```diff
@@ -22,41 +22,47 @@
 
 @(form:Form[_], packagings: Seq[Packaging])(implicit request: Request[_], messages: Messages,appConfig:AppConfig)
 
+@tableHeader = @{
+        if(packagings.size == 1)
```
Comment (kurtlogan) [inline app/views/goods_items_packagings.scala.html]: Sort styling out here

File: app/views/goods_items_origins.scala.html (hunk 4 above)
Comment (raghu1978) [reply app/views/goods_items_origins.scala.html]: the table header is becoming Any type and not. string

File: app/views/goods_items_origins.scala.html (hunk 6 above)
Comment (raghu1978) [reply app/views/goods_items_origins.scala.html]: its the label of buttons that are different and only in these 2 scenarios the fields needs to be displayed with different buttons

**Pull request merged**